backend/src/main/java/ai/verta/modeldb/project/FutureProjectServiceImpl.java

```diff
@@ -145,30 +145,95 @@ public void deleteProjectAttributes(
   @Override
   public void addProjectTags(
       AddProjectTags request, StreamObserver<AddProjectTags.Response> responseObserver) {
-    super.addProjectTags(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .addTags(request)
+              .thenCompose(
+                  unused -> futureProjectDAO.getPreAccessProjectById(request.getId()), executor)
+              .thenApply(
+                  project -> AddProjectTags.Response.newBuilder().setProject(project).build(),
+                  executor);
+      FutureGrpc.ServerResponse(responseObserver, response, executor);
+    } catch (Exception e) {
+      CommonUtils.observeError(responseObserver, e);
+    }
   }
 
   @Override
   public void getProjectTags(GetTags request, StreamObserver<GetTags.Response> responseObserver) {
-    super.getProjectTags(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .getTags(request)
+              .thenApply(tags -> GetTags.Response.newBuilder().addAllTags(tags).build(), executor);
+      FutureGrpc.ServerResponse(responseObserver, response, executor);
+    } catch (Exception e) {
+      CommonUtils.observeError(responseObserver, e);
+    }
   }
 
   @Override
   public void deleteProjectTags(
       DeleteProjectTags request, StreamObserver<DeleteProjectTags.Response> responseObserver) {
-    super.deleteProjectTags(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .deleteTags(request)
+              .thenCompose(
+                  unused -> futureProjectDAO.getPreAccessProjectById(request.getId()), executor)
+              .thenApply(
+                  project -> DeleteProjectTags.Response.newBuilder().setProject(project).build(),
```

Comment:
@hmacdonald-verta could you check with @convoliution that all of these APIs returning `Project` actually need them to return `Project`? I think this data is just discarded by the caller, so we could save some round trips here.